### PR TITLE
feat(alert, toast): pass arbitrary HTML attributes to host element

### LINKED ISSUE #23825: bug: expose role attribute on ion-alert + ion-toast
### Prequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [X] v5.x
- [ ] v6.x

### Current Behavior

While working with a community member, it was observed that there is no way to modify the `role` attribute for the `IonToast` and `IonAlert` components.

### Expected Behavior

As a developer, I would like to have the ability to set the `role` attribute as my application would like to set this attribute to `alert` for both components. 

`IonAlert` sets `role="dialog"` but in this application's accessibility context it needs to be set to "alert".

### Steps to Reproduce

Using the `AlertController` and `ToastController` there is no ability to manually set the `role` attribute.

### Code Reproduction URL

https://github.com/eric-horodyski/fw-ng-alert-toast-a11y

### Ionic Info

```
Ionic:

   Ionic CLI                     : 6.17.0 (/Users/eric/.volta/tools/image/packages/@ionic/cli/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 5.6.14
   @angular-devkit/build-angular : 12.1.4
   @angular-devkit/schematics    : 12.2.2
   @angular/cli                  : 12.1.4
   @ionic/angular-toolkit        : 4.0.0

Capacitor:

   Capacitor CLI      : 3.2.0
   @capacitor/android : not installed
   @capacitor/core    : 3.2.0
   @capacitor/ios     : not installed

Utility:

   cordova-res : 0.15.3
   native-run  : 1.4.0

System:

   NodeJS : v14.17.0 (/Users/eric/.volta/tools/image/node/14.17.0/bin/node)
   npm    : 7.14.0
   OS     : macOS Big Sur
```

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by willmartindev**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: resolves #23825 

## What is the new behavior?

- `ion-toast` and `ion-alert` now accept an `htmlAttributes` prop, which allows developers to pass arbitrary attributes to the host element. Potential use cases include overriding the default ARIA role or setting a custom `data-testid`.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by liamdebeasi — COMMENTED** [review]
For `AlertInput`, we allows developers to pass in an `attributes` object: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/alert/alert-interface.ts#L38

Should we do the same thing here to keep the API consistent?

**Comment by willmartindev** [comment]
> For `AlertInput`, we allows developers to pass in an `attributes` object: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/alert/alert-interface.ts#L38
> 
> Should we do the same thing here to keep the API consistent?

Yeah, I think that might be better. I can do that for both alert and toast.

File: core/src/components/alert/alert-interface.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ export interface AlertOptions {
   backdropDismiss?: boolean;
   translucent?: boolean;
   animated?: boolean;
+  additionalAttributes?: AlertAttributes;
```
Comment (liamdebeasi) [inline core/src/components/alert/alert-interface.ts:16]: Thoughts on calling this `htmlAttributes` similar to how we call it `cssClass`?

**Review by liamdebeasi — COMMENTED** [review]
The code looks good. We need to add tests to verify this behavior. Might want to add a spec test since we are testing what attributes get rendered on the host.

File: core/src/components/alert/alert-interface.ts (hunk 1 above)
Comment (liamdebeasi) [reply core/src/components/alert/alert-interface.ts:16]: Also I think JSX uses this same naming convention for attributes: https://github.com/ionic-team/ionic-framework/blob/7315e0157b917d2e3586c64f8082026827812943/core/src/components/radio/radio.tsx#L165

File: core/src/components/toast/readme.md [hunk 2]
```diff
@@ -405,6 +405,7 @@ export default defineComponent({
...
 | `enterAnimation` | --               | Animation to use when the toast is presented.                                                                                                                                                                                                                          | `((baseEl: any, opts?: any) => Animation) \| undefined` | `undefined` |
 | `header`         | `header`         | Header to be shown in the toast.                                                                                                                                                                                                                                       | `string \| undefined`                                   | `undefined` |
+| `htmlAttributes` | --               | Additional attributes to pass to the toast.                                                                                                                                                                                                                            | `ToastAttributes \| undefined`                          | `undefined` |
```
Comment (liamdebeasi) [inline core/src/components/toast/readme.md:408]: Please make sure you also update the `ToastOptions` and `AlertOptions` interfaces in the readmes

**Review by liamdebeasi — CHANGES_REQUESTED** [review]

**Review by liamdebeasi — APPROVED** [review]
Great job!

**Pull request merged**